modules/top-level/output.nix

```diff
@@ -258,21 +258,41 @@ in
             in
             ps: builtins.concatMap (f: f ps) deps;
 
-          # Combined plugin
+          getPropagatedAttrs = lib.filterAttrs (name: _: lib.strings.hasPrefix "propagated" name);
+
+          propagatedDependencies = lib.pipe toCombinePlugins [
+            (builtins.map getPropagatedAttrs)
+            (builtins.zipAttrsWith (name: lib.lists.concatLists))
+          ];
+
+          # Extract lua modules from propagated inputs
+          # as per toLuaModule
+          # https://github.com/NixOS/nixpkgs/blob/4f0dadbf/pkgs/development/lua-modules/lib.nix#L73-L85
+          luaModules = config.package.lua.pkgs.requiredLuaModules (
+            propagatedDependencies.propagatedBuildInputs or [ ]
+          );
```

Comment:
Hm, I was hoping this would help. Perhaps the `pathsToLink` argument is causing this to not work correctly?

Reply:
I've updated this to use the ``requiredLuaModules`` attr, but this still doesn't seem to be helping. Looking at the [nixpkgs lua wrapper](https://github.com/NixOS/nixpkgs/blob/6bc74d66a6d61f7c9dc1ed654764f577b4d265e9/pkgs/development/interpreters/lua-5/wrapper.nix#L32-L65), there may be some more work needed below in the "plugin pack" env package, such as using `buildLuaPath`?